federationsender/storage/interface.go

```diff
@@ -50,7 +50,18 @@ type Database interface {
 	GetPendingPDUServerNames(ctx context.Context) ([]gomatrixserverlib.ServerName, error)
 	GetPendingEDUServerNames(ctx context.Context) ([]gomatrixserverlib.ServerName, error)
 
+	// XXX: why don't these have contexts passed in?
```

Comment:
Largely because we want these things to happen regardless of what happens to the request context, so they instead generate their own contexts.